### ISSUE TITLE
AzureRM.Network 5.0.0 Cannot import module in Runbook

### ISSUE THREAD

**Issue by arcotek-ltd** [Azure/azure-powershell#5253]
### Cmdlet(s)
Import-Module AzureRm.Network

### PowerShell Version
Whatever version you have on your runbook servers? 5.1, the last time I checked.

### Module Version
5.0.0

### OS Version
Whatever version is on your runbook servers? Server 2012 (non-R2) I believe.

### Description
Import-Module AzureRM.Network fails. The runbook ends in a "Suspended" state with the error:
`The runbook job was attempted 3 times, but it failed each time.  Common reasons that runbook jobs fail can be found here:  https://docs.microsoft.com/en-us/azure/automation/automation-troubleshooting-automation-errors`

Following the link; suggests the error is caused because there are missing dependencies or the 400MB memory limit has been reached.

PowerShell Gallery states AzureRM.Network's has one dependency: AzureRM.Profile (>= 4.0.0). My automation account has 4.1.1. Assuming this is correct, I am lead to believe the problem is with this 400MB memory limit:

> Memory Limit. There are documented limits on how much memory allocated to a Sandbox Automation service limits so a job may fail it if it is using more than 400 MB of memory.

> 

Resource | Maximum Limit
-- | --
Max number of new jobs that can be submitted every 30 seconds per Automation Account (non Scheduled jobs) | 100
Max number of concurrent running jobs at the same instance of time per Automation Account (non Scheduled jobs) | 200
Max number of modules that can be imported every 30 seconds per Automation Account | 5
Max size of a Module | 100 MB
Job Run Time - Free tier | 500 minutes per subscription per calendar month
Max amount of memory given to a job | 400 MB
Max number of network sockets allowed per job | 1000

At this point, the runbook is failing and all the runbook is doing is `Import-Module AzureRM.Network`. If it's failing at this point, then I won't get very far with my runbook!

The automation account is the free tier.

I will log this as a ticket, but as far as I know, the 400MB limit is not a changeable value.

### Debug Output

I don't know how to get the process information for a runbook to determine how much memory it is using.

### Script/Steps for Reproduction

1. Make sure you have AzureRM.Network 5.0.0 installed in your Automation Account modules library.
2. Create PowerShell script runbook.
3. Add "Import-Module AzureRM.Network"
4. Run the runbook and review the output.

I'd be interested to see if others get the same result.

**Comment by markcowl** [comment]
@vrdmr I think this is mostly about how the versions of modules are updated in the portal blade.

**Comment by vrdmr** [comment]
@markcowl - I have reported this to the modules team. I'll try to get an update asap. Thanks.

**Comment by AnatoliB** [comment]
The provided repro steps are not sufficient, as I'm getting different results:

1. Created a new Automation account.
2. Imported AzureRM.Profile 4.1.1 from PS Gallery into the Automation account.
3. Imported AzureRM.Network 5.0.0 from PS Gallery into the Automation account.
4. Created a runbook (type: _PowerShell_) containing just `Import-Module AzureRm.Network`.
5. Started the runbook, and it successfully completed within a minute.
6. Updated all the Azure PS modules in the Automation account to the latest using the Update Azure Modules button in the portal, waited for the update to complete.
7. Started the runbook again, and it successfully completed within a minute.

Something else is going on. Please keep in mind that a job may crash for reasons not related to the content of this specific runbook or this specific module. For example, if you have jobs for other runbooks running in the same Automation account, these jobs will occasionally share the same sandbox process, and the memory limit is applied per sandbox, not per job. If another job crashes the sandbox, all jobs running in this sandbox will be affected. The complete set of modules imported into the Automation account also matters. So, in order to localize the issue, I would suggest running this on a clean Automation account first.

(By the way, invoking Import-Module from a PS runbook is normally not required in Azure Automation. After the module is imported into an Automation account, runbooks can simply start invoking commands from this module. But invoking Import-Module is not prohibited.)

**Comment by arcotek-ltd** [comment]
@AnatoliB 

There are no other runbooks running.
I have removed the Import-Module. As you say, it's no longer required to explicitly import required modules and the runbook completes without error, BUT:

In an attempt to identify where the issue is:

1. Create new automation account in West Central US
2. Create RunAs account.
3. Update all Azure modules.
4. Install AzureRM.Network (5.0.0) (Via "Browse Gallery").
5. Create a runbook with nothing other than Import-Module AzureRm.Network
6. Publish runbook
7. Turn on "Log verbose records" and "Log process records"
8. Start runbook

Exception:
`The runbook job was attempted 3 times, but it failed each time.  Common reasons that runbook jobs fail can be found here:  https://docs.microsoft.com/en-us/azure/automation/automation-troubleshooting-automation-errors`

Same error. The runbook also doesn't respond to "Stop". Which may be a portal bug as when I go back a blade, it allows me to start the runbook.

I've also turned off logging, rerun the runbook and get the same exception.
